### PR TITLE
Executing and Interacting with a Smart Contract (Part 2/4 of Hello World dApp Series)

### PR DESCRIPTION
**Pull request by nstrike2**
## Description

Added header tags for each of the titles as per request from @minimalsm. Please make sure to approve/merge [changes to Part 1/4](https://ethereum.org/en/developers/tutorials/hello-world-smart-contract/) to update its title before approving/merging this one.

## Related Issue

Continuation of https://github.com/ethereum/ethereum-org-website/pull/4158, which has now been closed due to inaccessibility.

Related Issue: https://github.com/ethereum/ethereum-org-website/issues/4068

### CODE REVIEW

**Comment by nstrike2** [comment]
@minimalsm @corwintines @pettinarip @samajammin @wackerow Can we get this merged? If not, happy to change what needs to be changed to get this merged.

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    "smart contracts",
+    "getting started",
+    "deploying"
+  ]
+skill: beginner
+lang: en
+sidebar: true
+published: 2021-10-06
+---
+# 💻 Executing and Interacting with a Smart Contract
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:20]: We don't currently have support for using Emojis in headings (we use Twemoji to ensure emojis are consistent across browsers but it breaks the table of contents include them in headers). Can we remove them across these tutorials, please?

```suggestion
# Executing and Interacting with a Smart Contract
```

Comment (minimalsm) [reply src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:20]: Also, could you please ensure all headers have header ids

e.g. `# Executing and Interacting with a Smart Contract {#executing-and-interacting}`

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+Before starting this tutorial on interacting with a smart contract, you should have completed part 1 — [Hello World Smart Contract](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract) (creating and deploying a smart contract). In part 3 we'll go over [submitting our contract to Etherscan](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract/submitting-your-smart-contract-to-etherscan) so anyone can understand how to interact with it!
+
+If you're more into video learning, click on the thumbnail below for a step-by-step walkthrough! 👇
+
+[![](https://img.youtube.com/vi/sQJ-XQBzEuc/maxresdefault.jpg)](https://www.youtube.com/watch?v=sQJ-XQBzEuc)
+
+And if you haven't already, you'll definitely need an Alchemy account to complete this tutorial as well as build anything on the blockchain. Sign up for a free account [here](https://alchemy.com/?a=eth-org-executing)!
+
+## Part 2: Interact with your Smart Contract
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:30]: Please add a header id here

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+**🤝 Step 3:** [Grab your contract ABI](./#step-3-grab-your-contract-ABI)
+
+**💻 Step 4:** [Create an instance of your contract](./#step-4-create-an-instance-of-your-contract)
+
+**📖 Step 5:** [Read the init message](./#step-5-read-the-init-message)
+
+**🔺 Step 6:** [Update the message](./#step-6-update-the-message)
+
+**⏫ Step 7:** [Read the new message](./#step-7-read-the-new-message)
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:50]: We already have a table of contents built into our tutorials page as a side panel, could we remove this please.

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+### Step 1: Create a interact.js file {#step-1-create-a-interact.js-file}
+
+This is the file where we'll write our interaction script. We'll be using the Ethers.js library that you previously installed in Part 1.&#x20;
+
+Inside your `scripts/`folder for the hardhat tutorial, or your home directory for the [Truffle tutorial](./#create-and-deploy-your-smart-contract-using-truffle), create a new file named `interact.js` add the following lines of code:
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:56]: The link here doesn't do anything?

```suggestion
Inside your `scripts/`folder for the Hardhat tutorial, or your home directory for the Truffle tutorial, create a new file named `interact.js` add the following lines of code:
```

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+And if you haven't already, you'll definitely need an Alchemy account to complete this tutorial as well as build anything on the blockchain. Sign up for a free account [here](https://alchemy.com/?a=eth-org-executing)!
+
+## Part 2: Interact with your Smart Contract
+
+Now that we've successfully deployed a smart contract to the ropsten network, let's test out our web3 skills and interact with it! We'll be using [Alchemy](https://alchemy.com/?a=interact) as our Ethereum API that lets us read and write to the blockchain.
+
+## 📋 Steps to execute and interact with your smart contract using Alchemy
+
+This guide assumes you already have an [Alchemy account](https://alchemy.com/?r=e68b2f77-7fc7-4ef7-8e9c-cdfea869b9b5), access to our [Dashboard](https://dashboard.alchemyapi.io/) and have gone through our [Hello World Smart Contract](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract) tutorial.
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:36]: As discussed in the previous PR, we need to remove all tracking links as it conflicts with our privacy policy.

```suggestion
This guide assumes you already have an [Alchemy account](https://alchemy.com/), access to our [Dashboard](https://dashboard.alchemyapi.io/) and have gone through our [Hello World Smart Contract](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract) tutorial. 
```

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+---
+# 💻 Executing and Interacting with a Smart Contract
+
+Before starting this tutorial on interacting with a smart contract, you should have completed part 1 — [Hello World Smart Contract](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract) (creating and deploying a smart contract). In part 3 we'll go over [submitting our contract to Etherscan](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract/submitting-your-smart-contract-to-etherscan) so anyone can understand how to interact with it!
+
+If you're more into video learning, click on the thumbnail below for a step-by-step walkthrough! 👇
+
+[![](https://img.youtube.com/vi/sQJ-XQBzEuc/maxresdefault.jpg)](https://www.youtube.com/watch?v=sQJ-XQBzEuc)
+
+And if you haven't already, you'll definitely need an Alchemy account to complete this tutorial as well as build anything on the blockchain. Sign up for a free account [here](https://alchemy.com/?a=eth-org-executing)!
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:28]:
```suggestion
And if you haven't already, you'll definitely need an Alchemy account to complete this tutorial as well as build anything on the blockchain. Sign up for a free account [here](https://alchemy.com/)!
```

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+// Contract
+const helloWorldContract = new ethers.Contract(CONTRACT_ADDRESS, contract.abi, signer);
+```
+
+You can read more about Providers, Signers, and Contracts in the [Ethers.js documentation](https://docs.ethers.io/v5/).
+
+### Step 5: Read the init message {#step-5-read-the-init-message}
+
+Remember when we deployed our contract with the`initMessage = "Hello world!"`? We are now going to read that message stored in our smart contract and print it to the console.&#x20;
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:144]:
```suggestion
Remember when we deployed our contract with the`initMessage = "Hello world!"`? We are now going to read that message stored in our smart contract and print it to the console.
```

File: src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+In JavaScript we use asynchronous functions to interact with networks. Check out this [article](https://blog.bitsrc.io/understanding-asynchronous-javascript-the-event-loop-74cd408419ff) to learn more about async functions.&#x20;
```
Comment (minimalsm) [inline src/content/developers/tutorials/executing-and-interacting-with-a-smart-contract/index.md:146]:
```suggestion
In JavaScript we use asynchronous functions to interact with networks. Check out this [article](https://blog.bitsrc.io/understanding-asynchronous-javascript-the-event-loop-74cd408419ff) to learn more about async functions.
```

**Review by minimalsm — CHANGES_REQUESTED** [review]

**Comment by nstrike2** [comment]
Hey @minimalsm , finished the changes for Part 2/4.

**Comment by nstrike2** [comment]
@minimalsm wanted to follow up to get these all pushed!

**Comment by nstrike2** [comment]
@minimalsm can we merge this? I'll be leaving Alchemy in 3 days to go back to school and then can't do much from here to continue on this after the holidays, so would love to have these be mutually beneficial to both you and us! Have made all the changes — should be ready to go.

**Comment by minimalsm** [comment]
Closing as per [this comment](https://github.com/ethereum/ethereum-org-website/pull/4284#issuecomment-1030104270)

**Pull request closed without merging**